Phosphor/src/edu/columbia/cs/psl/phosphor/instrumenter/SinkTaintingMV.java

```diff
@@ -25,50 +27,76 @@ public SinkTaintingMV(MethodVisitor mv, int access, String owner, String name, S
         this.owner = owner;
         this.name = name;
         this.desc = desc;
+        this.baseSink = BasicSourceSinkManager.getInstance().getBaseSink(owner, name, desc);
         this.isStatic = (access & Opcodes.ACC_STATIC) != 0;
         this.startLabel = new Label();
         this.endLabel = new Label();
     }
 
+    /* Adds code to make a call to enteringSink. */
+    private void callEnteringSink() {
+        super.visitFieldInsn(GETSTATIC, Type.getInternalName(Configuration.class), "autoTainter", Type.getDescriptor(TaintSourceWrapper.class));
+        super.visitLdcInsn(owner+"."+name+desc);
+        super.visitMethodInsn(INVOKEVIRTUAL, Type.getInternalName(TaintSourceWrapper.class), "enteringSink", "(Ljava/lang/String;)V", false);
+    }
+
+    /* Adds code to make a call to checkTaint for a non taint tag object.*/
+    private void callCheckTaintObject(int idx) {
+        super.visitFieldInsn(GETSTATIC, Type.getInternalName(Configuration.class), "autoTainter", Type.getDescriptor(TaintSourceWrapper.class));
+        super.visitVarInsn(ALOAD, idx);
+        super.visitLdcInsn(baseSink);
+        super.visitLdcInsn(owner+"."+name+desc);
+        super.visitMethodInsn(INVOKEVIRTUAL, Type.getInternalName(TaintSourceWrapper.class), "checkTaint", "(Ljava/lang/Object;Ljava/lang/String;Ljava/lang/String;)V", false);
+    }
+
+    /* Adds code to make a call to checkTaint for a taint tag*/
+    private void callCheckTaintTag(int tagIdx, int primitiveIdx, Type primitiveType) {
+        super.visitFieldInsn(GETSTATIC, Type.getInternalName(Configuration.class), "autoTainter", Type.getDescriptor(TaintSourceWrapper.class));
+        super.visitVarInsn(ALOAD, tagIdx);
+        super.visitVarInsn(primitiveType.getOpcode(ILOAD), primitiveIdx);
+        boxPrimitive(primitiveType);
```

Comment:
Instead of boxing to the java.lang.WrapperType, can we either pass the primitives directly or use a phosphor wrapper? the java.lang wrappers can introduce a higher overhead and risk of taint tag loss, since some of those classes can't be modified by phosphor at all (e.g. java.lang.character) - in which case we store the taint tag for that object in a hash map, which is a lot slower. Using the phosphor wrapper type guarantees that we keep the taint and value together.